src/plat/darwin/detect.c

```diff
@@ -351,12 +350,17 @@ void detect_wm(void)
 
 /*	detect_wm_theme
 	detects the theme associated with the WM detected in detect_wm().
-	On OS X, this will always be Aqua.
+	On OS X, there are dark and light theme, and various accent colours.
 */
 void detect_wm_theme(void)
 {
-	safe_strncpy(wm_theme_str, "Aqua", MAX_STRLEN);
-
+	char *accents[] = {"Graphite", "Red", "Orange", "Yellow", "Green", "", "Purple", "Pink", "Blue"};
+	Boolean e;
+	CFStringRef def = CFSTR(".GlobalPreferences");
+	CFIndex accent = CFPreferencesGetAppIntegerValue(CFSTR("AppleAccentColor"), def, &e);
+	CFIndex colorvar = CFPreferencesGetAppIntegerValue(CFSTR("AppleAquaColorVariant"), def, NULL);
+	CFPropertyListRef style = CFPreferencesCopyAppValue(CFSTR("AppleInterfaceStyle"), def);
+	snprintf(wm_theme_str, MAX_STRLEN, "%s %s", style?"Dark":"Light", e?accents[accent + 1]:colorvar == 6?"Graphite":"Blue");
```

Comment:
Nit: Use spaces to make ternary expressions easier to read. Also, use temporary variables for them here -- no need for them to be inlined.